scripts/deploy.sh

```diff
@@ -112,29 +143,42 @@ fi
 # Apply host security hardening with openstack-ansible-security
 # The is applied as part of setup-hosts.yml
 if [[ "$DEPLOY_HARDENING" == "yes" ]]; then
-  if grep -q '^apply_security_hardening:' /etc/openstack_deploy/user_variables.yml
-  then
-    sed -i "s/^apply_security_hardening:.*/apply_security_hardening: true/" /etc/openstack_deploy/user_variables.yml
+  if grep -q '^apply_security_hardening:' /etc/openstack_deploy/user_osa_variables_defaults.yml; then
+    sed -i "s/^apply_security_hardening:.*/apply_security_hardening: true/" /etc/openstack_deploy/user_osa_variables_defaults.yml
   else
-    echo "apply_security_hardening: true" >> /etc/openstack_deploy/user_variables.yml
+    echo "apply_security_hardening: true" >> /etc/openstack_deploy/user_osa_variables_defaults.yml
   fi
 else
-  if grep -q '^apply_security_hardening:' /etc/openstack_deploy/user_variables.yml
-  then
-    sed -i "s/^apply_security_hardening:.*/apply_security_hardening: false/" /etc/openstack_deploy/user_variables.yml
+  if grep -q '^apply_security_hardening:' /etc/openstack_deploy/user_osa_variables_defaults.yml; then
+    sed -i "s/^apply_security_hardening:.*/apply_security_hardening: false/" /etc/openstack_deploy/user_osa_variables_defaults.yml
   else
-    echo "apply_security_hardening: false" >> /etc/openstack_deploy/user_variables.yml
+    echo "apply_security_hardening: false" >> /etc/openstack_deploy/user_osa_variables_defaults.yml
   fi
 fi
 
 # ensure all needed passwords and tokens are generated
-./scripts/pw-token-gen.py --file /etc/openstack_deploy/user_secrets.yml
+./scripts/pw-token-gen.py --file /etc/openstack_deploy/user_osa_secrets_defaults.yml
 ./scripts/pw-token-gen.py --file $RPCD_SECRETS
 
 # Apply any patched files.
 cd ${RPCD_DIR}/playbooks
 openstack-ansible -i "localhost," patcher.yml
 
+# set permissions and lay down overrides files
+chown 0440 /etc/openstack_deploy/user_*_defaults.yml
```

Comment:
The secrets files can't be read-only unless their permissions are changed before and after pw-token-gen is run.